ksqldb-engine/src/test/java/io/confluent/ksql/statement/SourcePropertyInjectorTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.statement;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.confluent.ksql.config.SessionConfig;
+import io.confluent.ksql.parser.properties.with.CreateSourceAsProperties;
+import io.confluent.ksql.parser.properties.with.CreateSourceProperties;
+import io.confluent.ksql.parser.tree.CreateAsSelect;
+import io.confluent.ksql.parser.tree.CreateSource;
+import io.confluent.ksql.parser.tree.TableElements;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_INFERRED")
+@RunWith(MockitoJUnitRunner.class)
+public class SourcePropertyInjectorTest {
+
+  @Mock
+  private SessionConfig sessionConfig;
+
+  @Mock
+  private CreateSource createSource;
+  @Mock
+  private ConfiguredStatement<CreateSource> csStatement;
+  @Mock
+  private CreateSourceProperties originalCSProps;
+  @Mock
+  private CreateSourceProperties csPropsWithUnwrapping;
+  @Mock
+  private TableElements tableElements;
+  @Mock
+  private CreateSource csWithUnwrapping;
+
+  @Mock
+  private CreateAsSelect createAsSelect;
+  @Mock
+  private ConfiguredStatement<CreateAsSelect> csasStatement;
+  @Mock
+  private CreateSourceAsProperties originalCsasProps;
+  @Mock
+  private CreateSourceAsProperties csasPropsWithUnwrapping;
+  @Mock
+  private CreateAsSelect csasWithUnwrapping;
+
+  private SourcePropertyInjector injector;
+
+  @Before
+  public void setUp() {
+    when(csStatement.getStatement()).thenReturn(createSource);
+    when(csStatement.getSessionConfig()).thenReturn(sessionConfig);
+    when(createSource.getProperties()).thenReturn(originalCSProps);
+    when(createSource.getElements()).thenReturn(tableElements);
+    when(originalCSProps.withUnwrapProtobufPrimitives(true)).thenReturn(csPropsWithUnwrapping);
+    when(createSource.copyWith(tableElements, csPropsWithUnwrapping)).thenReturn(csWithUnwrapping);
```

Comment:
Maybe because nothing is mocked for `csWithUnwrapping`? You don't need to mock `csWithUnwrapping`?

Reply:
Hm, I don't think that's the problem. It's true that I don't need to mock `csWithUnwrapping` but `createSource.copyWith()` needs to be mocked and needs to return something, so I think this is a valid usage of a mock (otherwise I'd have to create a new `CreateSource` -- I guess it could be a stub but that should be the same thing from the perspective of findbugs).

Findbugs was specifically pointing to the lines `verify(originalCSProps).withUnwrapProtobufPrimitives(true);` and `verify(originalCsasProps).withUnwrapProtobufPrimitives(true);` as the offending lines, which is weird to me since it's valid Mockito usage to not care about the return value from `verify()`.

I spent some more time looking at this and couldn't figure it out so I ended up just removing the offending lines. The test coverage is still the same since the return value of `csWithUnwrapping` cannot be reached without going through the line I wanted to verify. (I only wanted to include the extra `verify()` check to make extra explicit for readers which method in the chain of method calls is the relevant one for the test.)